### PR TITLE
awx-ansible-modules: Add new job

### PR DESCRIPTION
**Pull request by Spredzy**

### CODE REVIEW

**Review by AlanCoding — APPROVED** [review]
looks good, thanks!

**Comment by AlanCoding** [comment]
You can see this target running in:

https://ansible.softwarefactory-project.io/logs/01/4701/e2e3b261fd0c2f3c9db68bce6228018c5f6e7b7c/check/awx-api/9073f34/ara-report/

```
cd /awx_devel
make prepare_test_modules
make[1]: Entering directory `/awx_devel'
rm -rf /awx_devel/awx_modules_test_venv
mkdir /awx_devel/awx_modules_test_venv
ln -s /usr/lib/python2.7/site-packages/ansible /awx_devel/awx_modules_test_venv/ansible
/venv/awx/bin/pip install --target=/awx_devel/awx_modules_test_venv git+https://github.com/ansible/tower-cli.git
Collecting git+https://github.com/ansible/tower-cli.git
  Cloning https://github.com/ansible/tower-cli.git to /tmp/pip-8zr1jtcd-build
Collecting click>=2.1 (from ansible-tower-cli==3.3.6)
  Downloading https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl (81kB)
Collecting colorama>=0.3.1 (from ansible-tower-cli==3.3.6)
  Downloading https://files.pythonhosted.org/packages/4f/a6/728666f39bfff1719fc94c481890b2106837da9318031f71a8424b662e12/colorama-0.4.1-py2.py3-none-any.whl
Collecting requests>=2.3.0 (from ansible-tower-cli==3.3.6)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl (57kB)
Collecting six>=1.7.2 (from ansible-tower-cli==3.3.6)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting PyYAML>=3.10 (from ansible-tower-cli==3.3.6)
  Cache entry deserialization failed, entry ignored
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.3.0->ansible-tower-cli==3.3.6)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
Collecting certifi>=2017.4.17 (from requests>=2.3.0->ansible-tower-cli==3.3.6)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/18/b0/8146a4f8dd402f60744fa380bc73ca47303cccf8b9190fd16a827281eac2/certifi-2019.9.11-py2.py3-none-any.whl (154kB)
Collecting idna<2.9,>=2.5 (from requests>=2.3.0->ansible-tower-cli==3.3.6)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests>=2.3.0->ansible-tower-cli==3.3.6)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/e6/60/247f23a7121ae632d62811ba7f273d0e58972d75e58a94d329d51550a47d/urllib3-1.25.3-py2.py3-none-any.whl (150kB)
Installing collected packages: click, colorama, chardet, certifi, idna, urllib3, requests, six, PyYAML, ansible-tower-cli
  Running setup.py install for ansible-tower-cli: started
    Running setup.py install for ansible-tower-cli: finished with status 'done'
Successfully installed PyYAML-5.1.2 ansible-tower-cli-3.3.6 certifi-2019.9.11 chardet-3.0.4 click-7.0 colorama-0.4.1 idna-2.8 requests-2.22.0 six-1.12.0 urllib3-1.25.3
You are using pip version 18.1, however version 19.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
============================= test session starts ==============================
platform linux -- Python 3.6.8, pytest-3.6.0, py-1.8.0, pluggy-0.6.0
Django settings: awx.settings.development (from ini file)
rootdir: /awx_devel, inifile: pytest.ini
plugins: xdist-1.27.0, timeout-1.3.3, pythonpath-0.7.3, mock-1.10.4, forked-1.0.2, django-3.5.1, cov-2.7.1, celery-4.3.0
collected 3 items

awx_modules/test/awx/test_job_template.py .x                             [ 66%]
awx_modules/test/awx/test_organization.py .                              [100%]

===================== 2 passed, 1 xfailed in 6.01 seconds ======================
make[1]: Leaving directory `/awx_devel'
```

**Comment by TristanCacqueray** [comment]
Note that the zuul logs are not properly streamed as we changed a setting that didn't get propagated to the static node. To fix that, we would need to run:
```yaml
- hosts: static-nodes
  tasks:
  - shell: "kill $(ps aux | grep ansible_module_zuul_console.py | awk '/python / { print $2 }')"
```

File: zuul.d/jobs.yaml [hunk 1]
```diff
@@ -45,3 +45,9 @@
     parent: awx-tests
     vars:
       command: /start_tests.sh detect-schema-change
+
+- job:
+    name: awx-ansible-modules
+    parent: awx-tests
+    vars:
+      command: /start_tests.sh prepare_test_modules
```
Comment (AlanCoding) [inline zuul.d/jobs.yaml]: @Spredzy After discussion in the related pull request with @elyezer https://github.com/ansible/awx/pull/4701#discussion_r324227177 we want to move toward a different name, `test_modules_all`.

Sorry about the churn.

Comment (ryanpetrello) [reply zuul.d/jobs.yaml]: For clarity, would it be better to use the term `collection` instead of `modules`?

Comment (AlanCoding) [reply zuul.d/jobs.yaml]: I guess it would be. The wording of "tower modules" is largely historical.

Comment (AlanCoding) [reply zuul.d/jobs.yaml]: `test_collection_all`

**Comment by Spredzy** [comment]
@TristanCacqueray done

**Comment by AlanCoding** [comment]
👍 all looks good to me

**Review by shanemcd — APPROVED** [review]

**Pull request merged**